docs/guides-adv_apicore.md

```diff
@@ -0,0 +1,102 @@
+  privateKey?: string
+  publicKey?: string
+  signingFunction?: signingFunction // Optional signing function.
+  url?: string // Optional. If present, overrides the url of the NEO node used
+  balance?: Balance // Optional. If present, this balance is used instead of retrieving a new one
+  intents?: TransactionOutput[]
+  sendingFromSmartContract?: boolean // This is used only if you are sending as a smart contract
+  claims?: Claims // Only used for claimGas
+  script?: string // Only used for doInvoke
+  gas?: number // Only used for doInvoke
+  override?: object // Transaction overrides
+}
+```
+
+## Information Retrieval
+
+This is the first step of every method. Based on the properties `net` and `address` provided, `neon-js` will attempt to retrieve the required information in order to construct the transaction.
+
+`net` represents the network which `neon-js` will attempt to work on. Usual terms provided are `MainNet` or `TestNet` which are networks hard-coded within the SDK itself. If you are using a private network, you will need to add a new network in settings or provide the direct url of your 3rd party provider here (NeonDB or neoscan).
+
+`address` is the address of the account that you wish to send from. This is the address which the assets originate from. This field is used to query your balance or claims from the 3rd party provider.
+
+Once the information is retrieved, it is appended to the configuration object under properties like `url`, `balance`, etc. and passed onto the next step.
+
+You may override any of these retrievals by populating the properties first. For example, if you wish to direct your calls to a specific NEO node, you should populate `url` with your intended url.
+
+## Transaction Building
+
+The `intents` property is used to specify the outputs of the transaction. Do note that the actual outputs will be a superset of the `intents` object due to leftover change when calculating.
+
+The `override` object is passed in to override any transaction properties after building the transaction.
+
+The information in the object is passed into the transaction building call and the built transaction is appended to the configuration under the property `tx`.
+
+## Transaction Signing
+
+For signing, the property `privateKey` from the configuration object is used to sign the object. An alternative to this is to provide a signing function under the property `signingFunction`.
+
+Instead of sending a user's private key to the config object, we can send the public key and a function that will sign the transaction.
+This function, the signingFunction, will receive the transaction and public key as parameters. Now, we can provide logic that retrieves the private key from our user - using the public key to do so - and signs the transaction when we retrieve the key.
+
+This the current way `neon-js` interacts with the Ledger Nano S for signing transactions.
+
+> Do note that the signing function has to return a Promise!
+
+```js
```

Comment:
Isn't it going to be confusing if we use ts and js interchangeably in the same doc?

Reply:
thats the only way i can present an interface. JS doesnt have this construct but honestly i would treat it more like pseudo code